pkg/storage/unified/resource/index_server.go

```diff
@@ -0,0 +1,138 @@
+package resource
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"google.golang.org/grpc"
+)
+
+type indexServer struct {
+	s     *server
+	index *Index
+	ws    *indexWatchServer
+}
+
+func (is indexServer) Search(ctx context.Context, req *SearchRequest) (*SearchResponse, error) {
+	res := &SearchResponse{}
+	return res, nil
+}
+
+func (is indexServer) History(ctx context.Context, req *HistoryRequest) (*HistoryResponse, error) {
+	return nil, nil
+}
+
+func (is indexServer) Origin(ctx context.Context, req *OriginRequest) (*OriginResponse, error) {
+	return nil, nil
+}
+
+func (is *indexServer) Index(ctx context.Context, req *IndexRequest) (*IndexResponse, error) {
+	if req.Key == nil {
```

Comment:
why the check for `req.Key == nil` ?

Reply:
if the key is nil then it will just reload the index.  if there is a key it will add the item (key, value) to the index